apriltags_ros/src/apriltag_detector.cpp

```diff
@@ -452,29 +493,71 @@ tf::Transform AprilTagDetector::getDepthImagePlaneTransform(const sensor_msgs::P
 
   pcl::PointIndices::Ptr polygonInlierIndices(new pcl::PointIndices());
 
+
+  // cannot register points to depth frame... we don't have depth so we are projecting to an unknown location in depth
+  //if(!same_frame_id_)
+  //{
+  //  ROS_DEBUG_THROTTLE(5.0, "Register polygon points from rgb to depth frame");
+  //  for(size_t i = 0; i < 4; i++)
+  //  {
+  //    polygon[i].first = (polygon[i].first - rgb_info->K[2]) / rgb_info->K[0] * depth_info->K[0] + depth_info->K[2];
+  //    polygon[i].second = (polygon[i].second - rgb_info->K[5]) / rgb_info->K[4] * depth_info->K[4] + depth_info->K[5];
+  //  }
+  //}
+
   pcl::PointCloud<pcl::PointXYZ> clipPolygon;
   clipPolygon.push_back(pcl::PointXYZ(polygon[0].first, polygon[0].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[1].first, polygon[1].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[2].first, polygon[2].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[3].first, polygon[3].second, 0));
   clipPolygon.push_back(pcl::PointXYZ(polygon[0].first, polygon[0].second, 0));
 
+  // generate the transform from depth_camera_reference frame to rgb_camera_reference frame
+  tf::Transform depth_to_rgb_transform;
+  string rgb_camera_frame_name = rgb_info->header.frame_id;
+  string depth_camera_frame_name = depth_info->header.frame_id;
+
+  getTransform(depth_camera_frame_name, rgb_camera_frame_name, depth_to_rgb_transform);
+
+  // generate a projection from rgb_camera_reference frame to image frame that is the P matrix in the rgb_info object
+
+  // actually convert the whole point cloud from depth frame to rgb frame
+
+  std::for_each(pointCloud->begin(), pointCloud->end(), [&depth_to_rgb_transform](pcl::PointXYZRGB & point) -> void {
+    tf::Vector3 pcl_point(point.x, point.y, point.z); pcl_point = depth_to_rgb_transform * pcl_point; point.x = pcl_point.x(); point.y = pcl_point.y(); point.z = pcl_point.z(); });
+
+  //depth_to_rgb_transform
+    //
+    Eigen::Matrix3f projection_mat;
```

Comment:
I think you can also do: Eigen::Matrix3f intrinsicMatrix(rgb_info->K, 3, 3);

Then you can do:
  
// https://github.com/6RiverSystems/apriltags_ros/blob/hotfix/simplify-transform/apriltags_ros/src/apriltag_detector.cpp#L471
pcl::transformPointCloud (*pointCloud, *pointCloud, depthToRgbTf); // from getTransform
pcl::transformPointCloud (*pointCloud, *pointCloud, intrinsicMatrix);

Can also combine the two transforms before the transformation so there is only one pass over the point cloud.

Reply:
Ohh I like that intrinsicMAtrix part :-)